src/borg/testsuite/helpers.py

```diff
@@ -800,7 +800,8 @@ def test_get_runtime_dir(monkeypatch):
     else:
         monkeypatch.delenv("XDG_RUNTIME_DIR", raising=False)
         monkeypatch.delenv("BORG_RUNTIME_DIR", raising=False)
-        assert get_runtime_dir() == os.path.join("/run/user", str(os.getuid()), "borg")
+        if not sys.platform.startswith("openbsd"):
+            assert get_runtime_dir() == os.path.join("/run/user", str(os.getuid()), "borg")
```

Comment:
i've seen some fix in platformdirs - can you re-test / adapt the test code with a current version?

Reply:
retested with platformdirs-3.8.0, and the issue is still there. Pull request still works for me, and is used [locally](https://github.com/openbsd/ports/blob/master/sysutils/borgbackup/2.0/patches/patch-src_borg_testsuite_helpers_py)

Reply:
well, the test would still fail, but it now returns a valid directory for openbsd i guess, so the test could be adapted to check that?

Reply:
Included with platformdirs-3.8.0 is https://github.com/platformdirs/platformdirs/commit/a02fb045afa46a3f2bc17359bdd77e7d537aa382, which causes `platformdirs.user_runtime_dir` to default to `/var/run/user/{getuid()}` on several BSD's, including OpenBSD.  `/var/run/user` is not part of OpenBS's base as shown by [hier(7)](http://man.openbsd.org/hier), and is therefore not a valid directory. Additionally, it seems that XDG's runtime directory is normally set by pam_systemd, which is not supported by OpenBSD.

I'm guessing that the recent change to platformdirs has not been tested on OpenBSD. Way forward is to discuss this with upstream of platformdirs.

I think it makes sense to address this issue in OpenBSD's port of borgbackup (until it has been resolved in platformdirs), and close this PR.

@ThomasWaldmann, What do you think?

Reply:
Guess getting `platformdirs` fixed would benefit a lot of software using that.
And if it is fixed there, no change in borg is needed.